addons/google_calendar/models/res_users.py

```diff
@@ -99,8 +99,8 @@ def stop_google_synchronization(self):
 
     def restart_google_synchronization(self):
         self.ensure_one()
-        if not self.google_cal_account_id:
-            self.google_cal_account_id = self.env['google.calendar.credentials'].sudo().create([{'user_ids': [Command.set(self.ids)]}])
+        if not self.google_calendar_account_id:
+            self.google_calendar_account_id = self.env['google.calendar.credentials'].sudo().create([{'user_ids': [Command.set(self.ids)]}])
```

Comment:
This line is a bit odd, doing the `create` should be enough as `user_ids` is the one2many to `google_calendar_account_id`. Or the goal was to check write access on `self` but it's a bit too hidden.

Reply:
same here :/

maybe @arj-odoo could explains more, I didn't refactor the code, just renamed google_cal_account_id to google_calendar_account_id.

Reply:
ps: need the sudo to create the acces_token, user only have the right to write on its own tokens.